applications/mne_analyze/mne_analyze/mainwindow.cpp

```diff
@@ -129,7 +128,16 @@ void MainWindow::closeEvent(QCloseEvent *event)
     m_pMultiView->saveSettings();
     saveSettings();
 
+    this->setAttribute(Qt::WA_DeleteOnClose);
+    m_pMenuView->setAttribute(Qt::WA_DeleteOnClose);
+
+    for(QDockWidget* widget : this->findChildren<QDockWidget*>()){
+        widget->setAttribute(Qt::WA_DeleteOnClose);
+        widget->close();
+    }
+
     emit mainWindowClosed();
+    QMainWindow::closeEvent(event);
 
     // default implementation does this, so its probably a good idea
     event->accept();
```

Comment:
I think we do not need this call anymore since we call `QMainWindow::closeEvent(event);` now